### PR TITLE
[Tablet Products] M1 Analytics

### LINKED ISSUE #12105: [Tablet Products] M1 Analytics
Ref: pfoUAQ-j0-p2

Add a new property horizontal_size_class (compact/regular) to the following events to track the split view usage:

- main_tab_products_selected (when the user selects the products tab)
- main_tab_products_reselected(when the user reselects the products tab)
- product_list_product_tapped (when the user taps on a product row in the products tab)
- product_list_add_product_button_tapped (when the user taps the + icon in the products tab to create a product)
- product_detail_loaded (when the product details form is shown to the user)

### PR DESCRIPTION
**Pull request by jaclync**
Closes: #12105

## Description

As in pfoUAQ-j0-p2, this PR adds a new property `horizontal_size_class` (`compact`/`regular`) to the following events to track the split view usage:

- main_tab_products_selected (when the user selects the products tab)
- main_tab_products_reselected(when the user reselects the products tab)
- product_list_product_tapped (when the user taps on a product row in the products tab)
- product_list_add_product_button_tapped (when the user taps the + icon in the products tab to create a product)
- product_detail_loaded (when the product details form is shown to the user)

## Screenshots

- [x] @jaclync tests on iPhone
- [x] @jaclync registers the event property in Tracks

Prerequisite: the device/simulator allows switching between expanded and collapsed states like a tablet

- In code, enable `splitViewInProductsTab` feature flag in `DefaultFeatureFlagService`
- Launch the app
- Go to the products tab --> in the console, `main_tab_products_selected` should be tracked with the correct `horizontal_size_class` property
- Tap on the products tab again --> in the console, `main_tab_products_reselected` should be tracked with the correct `horizontal_size_class` property
- Tap on a product in the product list --> in the console, `product_list_product_tapped` should be tracked with the correct `horizontal_size_class` property. after the product form is shown, `product_detail_loaded` should be tracked with the correct `horizontal_size_class` property
- From the product list, tap + to start the product creation flow --> in the console, `product_list_add_product_button_tapped` should be tracked with the correct `horizontal_size_class` property

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift [hunk 1]
```diff
@@ -114,7 +114,9 @@ final class AddProductCoordinator: Coordinator {
     func start() {
         switch source {
         case .productsTab:
-            analytics.track(event: .ProductsOnboarding.productListAddProductButtonTapped(templateEligible: isTemplateOptionsEligible))
+                analytics.track(event: .ProductsOnboarding
```
Comment (bozidarsevo) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift]: Looks like 1 extra tab in indentation?

**Review by bozidarsevo — APPROVED** [review]
🎉

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/188">17.6</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift]: Good catch, fixed in https://github.com/woocommerce/woocommerce-ios/pull/12109/commits/b3ac496d57afdb00ea5a322df190b93cf88d0c1e

**Pull request merged**